internal/storage/postgres/postgres.go

```diff
@@ -0,0 +1,131 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	t "github.com/repriest/url-shortener/internal/storage/types"
+	"time"
+)
+
+type pgStorage struct {
+	db *sql.DB
+}
+
+func NewPgStorage(dsn string) (t.Storage, error) {
```

Comment:
Здесь нет необходимости возвращать тип интерфейса. В идеале реализация вообще не обязана знать об этом интерфейсе, так как если тип реализует методы интерфейса он автоматически реализует сам интерфейс. При этом он будет реализовать не только интерфейс, который объявляет все методы, оно и какой-то один метод, Например:
```go
type BatchAppender interface {
	BatchAppend(entries []URLEntry) error
}
```

Возвращая интерфейс ты искусственно ограничиваешь клиентский код.
 
```suggestion
type PGStorage struct {
	db *sql.DB
}

func NewPgStorage(dsn string) (*PGStorage, error) {
```